parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputFormat.java

```diff
@@ -232,14 +240,36 @@ public ParquetInputFormat() {
   /**
    * @param configuration to find the configuration for the read support
    * @return the configured read support
+   * @deprecated use getReadSupportInstance static methods instead
    */
+  @Deprecated
   public ReadSupport<T> getReadSupport(Configuration configuration){
+    if (readSupportClass != null) {
+      return getReadSupportInstance(readSupportClass);
+    } else {
+      return getReadSupportInstance(configuration);
+    }
+  }
+
+  /**
+   * @param configuration to find the configuration for the read support
+   * @return the configured read support
+   */
+  @SuppressWarnings("unchecked")
+  public static <T> ReadSupport<T> getReadSupportInstance(Configuration configuration){
+    return getReadSupportInstance(
+        (Class<? extends ReadSupport>) getReadSupportClass(configuration));
+  }
+
+  /**
+   * @param readSupportClass to instantiate
+   * @return the configured read support
+   */
+  @SuppressWarnings("unchecked")
+  public static <T> ReadSupport<T> getReadSupportInstance(
```

Comment:
The problem I'm trying to fix here is the "weird caching" referred to in the comment below. Caching the `readSupportClass` could cause problems when reusing an instance of the `InputFormat`, which happens in HiveServer 2. I think the cleanest way to do this is to remove `getReadSupport` from the API and replace it with static methods.

Here is my reasoning behind making those public:
* `getReadSupportInstance(Configuration)` is used by the `mapred.DeprecatedParquetInputFormat` which means we either have to duplicate it or make it public in place of `getReadSupport`.
* `getReadSupportInstance(Class)` makes sense considering you could do this before by using:

```java
new ParquetInputFormat(MyReadSupport.class).getReadSupport()
```

These really just replace the method that I'm deprecating above.